skribenten-web/frontend/src/Brevredigering/LetterEditor/actions/switchFontType.ts

```diff
@@ -0,0 +1,293 @@
+
+  return hasSelectionAndMarkedText ? switchFontTypeOfMarkedText(args) : switchFontTypeOfCurrentWord(args);
+};
+
+const switchFontTypeForVariable = (args: {
+  draft: Draft<LetterEditorState>;
+  thisBlock: ParagraphBlock;
+  literalIndex: LiteralIndex;
+  fonttype: FontType;
+  variable: Draft<VariableValue>;
+}) => {
+  return newVariable({
+    id: args.variable.id,
+    text: args.variable.text,
+    fontType: args.fonttype,
+  });
+};
+
+const switchFontTypeOfMarkedText = (args: {
+  draft: Draft<LetterEditorState>;
+  thisBlock: ParagraphBlock;
+  literalIndex: LiteralIndex;
+  literal: Draft<LiteralValue>;
+  fonttype: FontType;
+}) => {
+  const selection = globalThis.getSelection()!;
+  const range = selection.getRangeAt(0);
+
+  const textBeforeTheSelection = (args.literal.editedText ?? args.literal.text).slice(0, range.startOffset);
+  const hasTextBeforeTheSelection = textBeforeTheSelection.length > 0;
+  const textInSelection = (args.literal.editedText ?? args.literal.text).slice(range.startOffset, range.endOffset);
+  const textAfterTheSelection = (args.literal.editedText ?? args.literal.text).slice(range.endOffset);
+  const hasTextAfterTheSelection = textAfterTheSelection.length > 0;
+
+  const changesTheWholeLiteral = textBeforeTheSelection.length === 0 && textAfterTheSelection.length === 0;
+
+  const newPreviousLiteral = newLiteral({
+    text: args.literal.text,
+    editedText: textBeforeTheSelection,
+    editedFontType: args.literal.editedFontType,
+  });
+  const newThisLiteral = newLiteral({
+    id: changesTheWholeLiteral ? args.literal.id : null,
+    text: args.literal.text,
+    editedText: changesTheWholeLiteral && args.literal.editedText === null ? null : textInSelection,
+    editedFontType: args.fonttype,
+    tags: changesTheWholeLiteral ? args.literal.tags : [],
+  });
+  const newNextLiteral = newLiteral({
+    text: args.literal.text,
+    editedText: textAfterTheSelection,
+    editedFontType: args.literal.editedFontType,
+  });
+
+  const newContent = [
+    ...(hasTextBeforeTheSelection ? [newPreviousLiteral] : []),
+    newThisLiteral,
+    ...(hasTextAfterTheSelection ? [newNextLiteral] : []),
+  ];
+
+  return {
+    resultingContent: newContent,
+    updatedCursorPosition: newThisLiteral.editedText?.length ?? 0,
+    newThisLiteralPositionInNewContent: hasTextBeforeTheSelection ? 1 : 0,
+    deletedContent: changesTheWholeLiteral ? null : args.literal.id,
+  };
+};
+
+/**
+ * Eksempel case er når dem har caret i midten av et ord, og vil bytte fonttypen på dette ordet.
+ */
+const switchFontTypeOfCurrentWord = (args: {
+  draft: Draft<LetterEditorState>;
+  thisBlock: ParagraphBlock;
+  literalIndex: LiteralIndex;
+  literal: Draft<LiteralValue>;
+  fonttype: FontType;
+}) => {
+  const cursorPosition = getCursorOffset();
+
+  if (cursorPosition < 0) {
+    return;
+  }
+
+  const text = args.literal.editedText ?? args.literal.text;
+
+  const isWhiteSpace = text[cursorPosition] === " ";
```

Comment:
Er det ei reell problemstilling her med tabs, newline og andre måtar å lage whitespace på, eller skjer ikkje det her?

Reply:
Vi kan enda ikke typisk tabbe for å lage større avstander i brevet. Dette vil seff være et problem hvis det blir implementert på en eller måte. Når vi lager newline, så lager vi en helt som følger blocken vi var på. Så da finnes det ikke noe bak siste bokstaven i ordet.